net/tstun/wrap.go

```diff
@@ -481,6 +494,16 @@ func (t *Wrapper) filterIn(buf []byte) filter.Response {
 				f(data)
 			}
 		}
+	case ipproto.UDP:
+		if pkt := p.Payload(); disco.LooksLikeDiscoWrapper(pkt) {
+			discoFrom := pkt[len(disco.Magic):][:len(tailcfg.DiscoKey{})]
+			if selfDiscoPub, ok := t.discoKey.Load().(tailcfg.DiscoKey); ok {
+				if bytes.Equal(selfDiscoPub[:], discoFrom) {
+					t.logf("[unexpected] received self disco package over tstun; dropping")
```

Comment:
To confirm, when issue 1526 reproduces, would we expect to see one of these log lines, or one per packet send (for other reasons)? I know we have a rate-limiter, but even going through the rate-limited once per packet is going to be not cheap. Not that I see any good alternatives.

Is there any other info we want as part of debugging this? (I'd suggest remote port+IP but I guess that has PII problems.)

Reply:
I think that stopping one will stop the flood.

But we'll see. We'll have the users test it before the release and we can adjust.